addons/website/models/website.py

```diff
@@ -612,26 +612,51 @@ def _render_snippet(key):
             snippets_cache[key] = data
             return data
 
+        text_generation_target_lang = self.get_current_website().default_lang_id.code
+        nb_terms_translated, nb_terms_not_translated = 0, 0
         generated_content = {}
         for page_code in requested_pages - {'privacy_policy'}:
             snippet_list = snippet_lists.get(page_code, [])
             for snippet in snippet_list:
                 render, placeholders = _render_snippet(f'website.{snippet}')
+                if text_generation_target_lang != 'en_US':
+                    # Check if terms are translated.
+                    translation_dictionary = self.env['website.page']._fields['arch_db'].get_translation_dictionary(
+                        IrQweb._render(f'website.{snippet}', cta_data, lang="en_US").__str__(),
+                        {text_generation_target_lang: render.__str__()},
+                    )
+                    for from_lang_term, to_lang_terms in translation_dictionary.items():
+                        if from_lang_term == to_lang_terms[text_generation_target_lang]:
+                            nb_terms_not_translated += 1
+                        else:
+                            nb_terms_translated += 1
                 for placeholder in placeholders:
                     generated_content[placeholder] = ''
-        try:
-            response = self._OLG_api_rpc('/api/olg/1/generate_placeholder', {
-                'placeholders': list(generated_content.keys()),
-                'lang': website.default_lang_id.name,
-                'industry': industry,
-            })
-            name_replace_parser = re.compile(r"XXXX", re.MULTILINE)
-            for key in generated_content:
-                if response.get(key):
-                    generated_content[key] = (name_replace_parser.sub(website.name, response[key], 0))
-        except AccessError:
-            # If IAP is broken continue normally (without generating text)
-            pass
+        if text_generation_target_lang != 'en_US':
+            translated_ratio = nb_terms_translated / (nb_terms_not_translated + nb_terms_translated)
+            logger.info("Number of not translated terms: %s", nb_terms_not_translated)
+            logger.info("Number of translated terms: %s", nb_terms_translated)
+        else:
+            translated_ratio = 1
+
+        logger.info("Ratio of translated content for %s: %s", text_generation_target_lang, translated_ratio)
```

Comment:
I think we don't need a difference for english or not in terms of log.
- I don't know if there is any convention but I would not log this info. Maybe use `logger.debug` instead so that you can have the log when in debug mode but not all the time
- Instead of 3 logs, only keep the ratio one, and you can put something like "55% (11/20)" (if 11 translated over 20 for example).